### ISSUE TITLE
Bitbucket data center test report

### ISSUE THREAD

**Issue by malomarrec** [sourcegraph/sourcegraph#29915]
TL;DR: all tested functionalities works. Setup and docs are sometimes unclear.

My primary goal is to verify that Batch Changes is compatible with Bitbucket Data Center, but I noted other UX details along the way.

1) I was able to connect BBDC to Sourcegraph, using the "Bitbucket server" code host option.
Friction: it was unclear if I had to use Bitbucket.org or Bitbucket server as the codehost type. After trying out both, I realized Server made more sense.
The permissions required were clear.

<img width="979" alt="Screenshot 2022-01-19 at 15 24 10" src="https://user-images.githubusercontent.com/25070988/150151578-b6607921-4333-4674-b6fe-f3b08055a60e.png">

2) I added a few repositories, stored under the `DEMO` [project](https://bitbucket-dc.sgdev.org/projects/DEMO) on BBDC
At first I was surprised to see repos appear as `DEMO/repo-name` but it made sense afterwards https://demo.sourcegraph.com/bitbucket-dc.sgdev.org/DEMO/bbdc-traefik

3) I wanted to test out Batch Changes, so I started setting up [webhooks](https://docs.sourcegraph.com/admin/external_service/bitbucket_server). I landed on this [page](https://docs.sourcegraph.com/integration/bitbucket_server#sourcegraph-bitbucket-server-plugin) about the BB server plugin, then on this [repo](https://github.com/sourcegraph/bitbucket-server-plugin/tree/master).
- Reading the README was confusing because it mentions "external services" and I had not encountered that word before (I was setting up a "codehost connection"). Is that the same?
- I added `corsOrigin` to demo.sourcegraph.com as instructed
After a few steps, I landed [here](https://github.com/sourcegraph/bitbucket-server-plugin/tree/master/src/main/java/com/sourcegraph/webhook) hoping to find instructions on how to setup webhooks, with no luck. Maybe the plugin repo could provide a better next step in the README.   After erring a little bit, I went back to the docs and eventually landed on: https://docs.sourcegraph.com/admin/external_service/bitbucket_server#webhooks
- The docs keep referring to "Site admin > Manage repositories" which does not exist. I realized this is actually "Site admin > Manage Codehosts", but Cmd+F-ing around does not work.
- Otherwise, the docs page was very helpful!

At this stage, I think I'm done and can use Bacth Changes. Let's find out! Hello world time.

Arg. Sounds like I'm missing creds? Sure actually, I'm a site-admin and this is personal. Love this little in-product creds pop-up by the way, it's much easier to use than the previous steps where I had to toggle between several site admin pages, bitbucket, BB docs, and Sourcegraph docs.
![image](https://user-images.githubusercontent.com/25070988/150155829-4b8ee24c-d8ae-4e79-9ed0-0edf192ab3aa.png)

Done. Let's apply! Let's publish! It worked, so far so good.

- [x] apply / publish
- [x] merge / close (sync instantaneous on 4 repos)
- [x] bulk comment
- [x] bulk merge
- [x] bulk close
- [x] re-apply and re-open closed changesets / no-op on merged

--------

A few UI glitches unrelated to Bitbucket data center still worth noting:
- I tried deleting a changeset on BB DC (which is not the same as "decline").  When doing so, nothing happens in the batch change (the changesets shows as "open", clicking on the link results in a 404). [edit]: after triggering a sync manually, it actually works.
- After re-applying a batch change, I get this view with 3 changesets out of 4. Refreshing displays 4
![image](https://user-images.githubusercontent.com/25070988/150158171-60d90da9-e004-4020-8aca-e1a774575122.png)
